qulice-findbugs/src/test/java/com/qulice/findbugs/FindBugsExcludesTest.java

```diff
@@ -0,0 +1,262 @@
+ * products derived from this software without specific prior written
+ * permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT
+ * NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.qulice.findbugs;
+
+import com.qulice.spi.Environment;
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Iterator;
+import org.dom4j.Document;
+import org.dom4j.io.SAXReader;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.collection.IsEmptyCollection;
+import org.hamcrest.core.IsEqual;
+import org.hamcrest.core.IsNot;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+/**
+ * Test for exclusion filters.
+ * @checkstyle JavadocMethodCheck (500 lines)
+ * @checkstyle LineLengthCheck (500 lines)
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ * @since 0.18.15
+ */
+@SuppressWarnings({
+    "PMD.TooManyMethods",
+    "PMD.AvoidDuplicateLiterals"
+    })
+class FindBugsExcludesTest {
+
+    @Test
+    public void ifCantCreateTempFile() {
+        final Environment env = Mockito.mock(Environment.class);
+        Mockito.when(env.tempdir()).thenReturn(new File("/dev/null"));
+        Mockito.when(env.excludes("findbugs")).thenReturn(
+            Arrays.asList("org.package.Class")
+        );
+        Assertions.assertThrows(
+            IllegalStateException.class,
+            new FindBugsExcludes(env)::asArguments,
+            "Didn't throw an exception for invalid file"
+        );
+    }
+
+    @Test
+    public void noExcludesNoArguments() throws IOException {
+        final Environment env = new Environment.Mock()
+            .withExcludes("");
+        MatcherAssert.assertThat(
+            "The arguments were not empty for empty excludes",
+            new FindBugsExcludes(env).asArguments(),
+            new IsEmptyCollection<>()
+        );
+    }
+
+    @Test
+    public void notEmptyExcludesShouldBeNotEmptyArguments() throws Exception {
+        final Environment env = new Environment.Mock()
+            .withExcludes("Foo,Bar");
+        final FindBugsExcludes excludes = new FindBugsExcludes(env);
+        final Collection<String> arguments = excludes.asArguments();
```

Comment:
@ilyakharlamov I think that you don't need to create the `exclude` variable here